### PR TITLE
[ML] Remove allocators from cache after usage

### PR DESCRIPTION
**Pull request by valeriy42**
In the `CBoostJsonWriteBase` class, we maintain a cache of allocators that should be reused when we write new results. Although we took measures to free memory in these allocators, it was still not completely possible. As a result, we noticed a gradual increase in memory usage by native processes due to the increasing size of the allocators in the cache.

This PR ensures that the allocator is removed from the cache and the memory is freed.

Related to #2664

### CODE REVIEW

**Comment by valeriy42** [comment]
buildkite run_qa_tests on linux

File: include/core/CBoostJsonWriterBase.h [hunk 1]
```diff
@@ -134,6 +131,14 @@ class CBoostJsonWriterBase {
         return this->getAllocator()->get();
     }
 
+    void removeAllocator(const std::string& allocatorName) {
+        if (m_AllocatorCache.find(allocatorName) != m_AllocatorCache.end()) {
+            TPoolAllocatorPtr allocator = m_AllocatorCache[allocatorName];
```
Comment (jan-elastic) [inline include/core/CBoostJsonWriterBase.h:136]: It looks slightly inefficient to do `cache.find(key)` followed by `cache[key]` and `erase(key)`. It does the map lookup 3x.

Why not do:
```
allocator = cache.find(name);
if (allocator != cache.end()) {
  allocator->reset();
  cache.erase(allocator);
}
```

**Review by jan-elastic — APPROVED** [review]
LGTM

File: include/core/CBoostJsonWriterBase.h (hunk 1 above)
Comment (valeriy42) [reply include/core/CBoostJsonWriterBase.h:136]: Good point. I adjusted the code.

**Pull request merged**